migration/sql-files/098-boards.sql

```diff
@@ -0,0 +1,33 @@
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+
+-- new tables for the board and boardcolumn entities
+CREATE TABLE work_item_boards (
+    created_at timestamp with time zone,
+    updated_at timestamp with time zone,
+    deleted_at timestamp with time zone,
+    id uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+    version integer DEFAULT 0 NOT NULL,
+    space_template_id uuid REFERENCES space_templates(id) ON DELETE CASCADE,
+    name text NOT NULL CHECK(name <> ''),
+    description text NOT NULL CHECK(description <> ''),
+    context_type text NOT NULL CHECK(context_type <> ''),
+    context text NOT NULL CHECK(context <> ''),
+    CONSTRAINT work_item_board_name_space_template_id_unique UNIQUE(space_template_id, name)
+);
+
+CREATE INDEX work_item_boards_space_template_uidx ON work_item_boards (space_template_id) WHERE deleted_at IS NULL;
+
+CREATE TABLE work_item_board_columns (
+    created_at timestamp with time zone,
+    updated_at timestamp with time zone,
+    deleted_at timestamp with time zone,
+    id uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+    board_id uuid REFERENCES work_item_boards(id) ON DELETE CASCADE,
+    name text NOT NULL CHECK(name <> ''),
+    column_order integer DEFAULT 0 NOT NULL,
+    trans_rule_key text NOT NULL CHECK(trans_rule_key <> ''),
+    trans_rule_argument text NOT NULL CHECK(trans_rule_argument <> ''),
+    CONSTRAINT work_item_board_column_name_board_id_unique UNIQUE(board_id, name)
```

Comment:
Shouldn't we have a unique constraint on `column_order + board_id`?
Each board should have unique `column_orders`

Reply:
@jarifibrahim I'm on it.

Reply:
@jarifibrahim by adding this I found out that the order was incorrectly parsed from the template. 

@michaelkleinhenz every order was 0 because the type field on the `BoardColumn` struct was specifying `column_order` as it's json struct tag. But that was not used in all the places of the space templates. Sometimes it was `order` and sometimes it was `column_order`. I've changed it to be `order` everywhere and added a test for the uniqueness of board_id+column order in 3374165e152dd3a38bd44da8419c9862e14654f6.

@jarifibrahim this was a really good catch!!!!